### PR TITLE
Add [:each | each key matchesRegex: '^123$']  support

### PR DESCRIPTION
**Pull request by zecke**
Add MQField>>#matchesRegex:, add MQRegexCondition and test case. Sadly a ton of changes I have no way to influence (or what can I do to have the change reduced to what I actually modified?)

### CODE REVIEW

**Comment by zecke** [comment]
@noha What do you think about this protocol?

**Comment by estebanlm** [comment]
Is ok for me. 
Don't we need a `matchesRegexIgnoreCase:` ?

**Comment by noha** [comment]
I don't know if it looks ok because I have trouble finding the code through all the diffs in metadata. How do you guys read the diff? From the image?

**Comment by estebanlm** [comment]
With a lot of patience :)

Yes is annoying, that's why I support Dale's suggestion (switch to metadata-less), problem is that it requires latest filetree versions then people using (to develop) Pharo 3 and 4 will have some problems. 

But is a problem just for developers, users will still be able to load... is just that they will have packages with the form `Package.cypress.1`

File: mc/Mongo-Queries.package/MQRegexCondition.class/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Check if $regex is matching
```
Comment (zecke) [inline mc/Mongo-Queries.package/MQRegexCondition.class/README.md]: Class comment

File: mc/Mongo-Queries.package/MQRegexCondition.class/instance/label.st [hunk 2]
```diff
@@ -0,0 +1,3 @@
+as yet unclassified
+label
+	^ '$regex'
```
Comment (zecke) [inline mc/Mongo-Queries.package/MQRegexCondition.class/instance/label.st]: Verb to use for MongoDB

File: mc/Mongo-Queries.package/MQField.class/instance/matchesRegex..st [hunk 3]
```diff
@@ -0,0 +1,6 @@
+comparing
```
Comment (zecke) [inline mc/Mongo-Queries.package/MQField.class/instance/matchesRegex..st]: Implementation of [:each | each field matchesRegex: string ]

File: mc/Mongo-Tests-Queries.package/MQQueryTest.class/instance/testMatchesRegex.st [hunk 4]
```diff
@@ -0,0 +1,6 @@
+tests
+testMatchesRegex
```
Comment (zecke) [inline mc/Mongo-Tests-Queries.package/MQQueryTest.class/instance/testMatchesRegex.st]: actually testcase to try to match something

**Comment by zecke** [comment]
Any guide to have less metadata? This was with a recent Pharo5 image.

For ignore case ... we have a slight issue in protocol. For mongodb we would pass the regexp.. and then use /i to indicate we want to ignore the case.. for pharo that doesn't work.. we would need to modify the argument in the mongotalk case. Is this something we want?

File: mc/Mongo-Queries.package/.filetree [hunk 5]
```diff
@@ -1,5 +1,4 @@
 {
-	"separateMethodMetaAndSource" : false,
 	"noMethodMetaData" : true,
-	"useCypressPropertiesFile" : true
-}
\ No newline at end of file
+	"separateMethodMetaAndSource" : false,
+	"useCypressPropertiesFile" : true }
```
Comment (tinchodias) [inline mc/Mongo-Queries.package/.filetree]: did this change have any effect?

**Comment by tinchodias** [comment]
The noise in the diffs because of the metadata is impressive, but @zecke highlighted the important parts and looks fine IMHO (but I don't have experience in this part of Mongo so I wait for others)

**Pull request merged**